adal/src/main/java/com/microsoft/aad/adal/AuthenticationContext.java

```
@@ -726,6 +726,93 @@ public void onError(Exception exc) {
+        final APIEvent apiEvent = createApiEvent(mContext, clientId, requestId, EventStrings.ACQUIRE_TOKEN_SILENT_SYNC);
+        apiEvent.setPromptBehavior(PromptBehavior.Auto.toString());
+
+        final AuthenticationRequest request = new AuthenticationRequest(mAuthority, resource,
+                clientId, userId, getRequestCorrelationId(), getExtendedLifetimeEnabled(), forceRefresh);
+        request.setSilent(true);
+        request.setPrompt(PromptBehavior.Auto);
+        request.setUserIdentifierType(UserIdentifierType.UniqueId);
+        request.setTelemetryRequestId(requestId);
+
+        final Looper currentLooper = Looper.myLooper();
+        if (currentLooper != null && currentLooper == mContext.getMainLooper()) {
+            Logger.e(TAG + methodName,
+                    "Sync network calls must not be invoked in main thread. "
+                            + "This method will throw android.os.NetworkOnMainThreadException in next major release",
+                    new NetworkOnMainThreadException());
+        }
+        createAcquireTokenRequest(apiEvent).acquireToken(null, false, request,
+                new AuthenticationCallback<AuthenticationResult>() {
+                    @Override
+                    public void onSuccess(AuthenticationResult result) {
+                        authenticationResult.set(result);
+                        latch.countDown();
+                    }
+
+                    @Override
+                    public void onError(Exception exc) {
+                        exception.set(exc);
+                        latch.countDown();
+                    }
+                });
+
+        latch.await();
+
+        Exception e = exception.get();
+        if (e != null) {
+            if (e instanceof AuthenticationException) {
+                throw (AuthenticationException) e;
+            } else if (e instanceof RuntimeException) {
+                throw (RuntimeException) e;
```

Is there any sense in wrapping this inside of an `AuthenticationException` to rethrow? Are we expecting devs to catch instances of `RuntimeException` or let it crash? (This was TBD, as I recall)

---

This is existing pattern.  I'll take a look.